clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java

```diff
@@ -158,8 +157,7 @@ public Password getPassword(String key) {
     }
 
     /**
-     * Put all keys that do not start with {@code prefix} and their parsed values in the result map and then
-     * put all the remaining keys with the prefix stripped and their parsed values in the result map.
+     * Put all keys that start with {@code prefix} with the prefix stripped and their parsed values in the result map.
```

Comment:
The method doesn't process the keys that don't start with the prefix. So, I have updated the java doc.